warehouse/migrations/versions/16cafc7e36d6_create_identities_table.py

```diff
@@ -0,0 +1,172 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+create identities table
+
+Revision ID: 16cafc7e36d6
+Revises: bd71566c2877
+Create Date: 2022-11-03 18:03:26.634280
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+revision = "16cafc7e36d6"
+down_revision = "bd71566c2877"
+
+
+def upgrade():
```

Comment:
Flagging for review: I'm a little murky on why these migrations were generated -- my understanding of the changes I've made is that they shouldn't have affected any existing rows, since the inheritance model being introduced is "concrete" instead of joined or single-table.

Reply:
The main difference I see between the upgrade and downgrade methods is the usage of the 'public' namespace.

Reply:
Curious if you happened to generate these on a Postgres 15 instance, since the public namespace behaviors changes a bit

Reply:
I generated it using whatever the docker-compose configuration for Warehouse uses, so it _should_ be 14.4:

https://github.com/pypi/warehouse/blob/606bce63472a7979eff0a37ea8c44c335b9c118f/docker-compose.yml#L27-L35

Reply:
Thanks for clarifying! helpful for diagnosis. I _think_ I've gotten to the bottom of why this happens.

The development database is a `pg_dump` from a point in time from test.pypi.org, and the process to refresh the dev db [is documented here](https://warehouse.pypa.io/development/development-database.html).
The last such refresh was made in #5096 back in November 2018 using:

```shellsession
$ xzgrep -- "-- Dumped" dev/example.sql.xz
-- Dumped from database version 10.5
-- Dumped by pg_dump version 10.5
```

I dug a bit into Postgres' changes to the `pg_dump` behavior vis a vis `public` schema, since the migration shows a change in lookup path for the `gen_random_uuid` function that comes with a plpgsql extension - the closest I came to is https://github.com/postgres/postgres/commit/5955d934194c3888f30318209ade71b53d29777f - but it doesn't look like that changes much, since even a `pg_dump` with the latest version 14 includes some records of `

If we `pg_dump` the current dev database version after `make initdb`, we get a mix of with and without the `public` schema:

```shellsession
grep gen_random_uuid dump-14.sql | sort | uniq -c
  32     id uuid DEFAULT gen_random_uuid() NOT NULL,
  10     id uuid DEFAULT public.gen_random_uuid() NOT NULL,
```
confirming it's not a change in `pg_dump` behavior.

This hasn't changed since 2015:
https://github.com/pypi/warehouse/blob/c2a0c894bd0380d9df5c0973591e7acc343c919d/warehouse/db.py#L91

@ewdurbin Maybe we need a fresh db dump for dev? The `dev/clean.sql` file noted in the process no longer exists as of [#5882 ](https://github.com/pypi/warehouse/pull/5882/files#diff-af9fea57764b3a0e3be63f5c3d5f2996f220a4644f413b14efde034ea63f4954) - do we need to bring it back or update the process?

Reply:
tracked kinda in https://github.com/pypi/warehouse/issues/6504

Reply:
1387d3a1212cb56feb114611cf44739f63707eb2 should resolve this, thanks @ewdurbin! I'll rebase later today.